src/ui/view/share/data/dropDown/dropDown.tsx

```diff
@@ -0,0 +1,157 @@
+import type { FC } from 'react'
+import { useMemo, useCallback, useState, useEffect } from 'react'
+import { Collapsible } from '@/ui/component/collapsible/collapsible'
+import { SearchBar } from '@/ui/component/searchbar/searchbar'
+import { isSubstringOf } from '@/util/util'
+import { DynamicCheckbox } from '@/ui/view/dataverse/component/dynamicCheckbox/dynamicCheckbox'
+import { NoResultFound } from '@/ui/view/dataverse/component/noResultFound/noResultFound'
+import { Tag } from '../../../../component/tag/tag'
+import './dropDown.scss'
+import { DROPDOWN_MAX_SEARCH_RESULTS, DROPDOWN_TAGS_DISPLAY_LIMIT } from '@/ui/config'
+
+export type DropDownOption = {
+  value: string
+  label: string
+  disabled?: boolean
+  selected?: boolean
+}
+
+type SelectionItemType = 'checkbox'
+
+type DropDownProps = {
+  filterName: string
+  filterValues: string[]
+  searchPlaceholder: string
+  selectionType: SelectionItemType
+  maxSearchResults?: number
+  onSelectionChange: (values: string[]) => string[]
+}
+
+// eslint-disable-next-line max-lines-per-function
+export const DropDown: FC<DropDownProps> = ({
+  filterName,
+  filterValues,
+  searchPlaceholder,
+  selectionType,
+  maxSearchResults = DROPDOWN_MAX_SEARCH_RESULTS,
+  onSelectionChange
+}) => {
+  const createDefaultFilterOption = (value: string): DropDownOption => ({
+    label: filterName,
+    value
+  })
+
+  const [searchTerm, setSearchTerm] = useState<string>('')
+
+  const [filterOptions, setFilterOptions] = useState<DropDownOption[]>(
+    filterValues
+      .map(value => createDefaultFilterOption(value))
+      .sort((a, b) => a.value.localeCompare(b.value))
+  )
+
+  const handleSearch = useCallback(
+    (searchTerm: string) => {
+      setSearchTerm(searchTerm)
+    },
+    [setSearchTerm]
+  )
+
+  const handleSelectedChange = useCallback(
+    (changedOption: Pick<DropDownOption, 'selected' | 'value' | 'label'>) => {
+      setFilterOptions(prevFilterOptions =>
+        prevFilterOptions.map(option =>
+          option.value === changedOption.value
+            ? { ...option, selected: changedOption.selected }
+            : option
+        )
+      )
+    },
+    [setFilterOptions]
+  )
+
+  const foundFilterOptions = useMemo(
+    () =>
+      searchTerm.trim() === ''
+        ? filterOptions
+        : filterOptions
+            .filter(({ value }) => isSubstringOf(searchTerm, value))
+            .slice(0, maxSearchResults),
+    [filterOptions, searchTerm, maxSearchResults]
+  )
+
+  const selectedOptions = useMemo(
+    () => filterOptions.filter(({ selected }) => selected),
+    [filterOptions]
+  )
+
+  useEffect(() => {
+    if (selectedOptions.length) {
+      onSelectionChange(selectedOptions.map(({ value }) => value))
+    }
+  }, [onSelectionChange, selectedOptions])
```

Comment:
Useless, `onSelectionChange` callback should be called in the `handleSelectedChange` function.